soc/xtensa/intel_adsp/cavs/power.c

```diff
@@ -180,6 +180,52 @@ void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 }
 #endif /* CONFIG_PM */
 
+#ifdef CONFIG_ARCH_CPU_IDLE_CUSTOM
+/* xt-clang removes any NOPs more than 8. So we need to set
```

Comment:
```suggestion
/* xt-clang removes any NOPs more than 8. So we need to set
```

Please provide at least one xt-lang version that with confirmed reproduction. This workaround should refer to a proper issue number but for now the mere output of `xt-lang --version` will already be infinitely better than nothing.

If there was a past discussion in https://github.com/thesofproject/sof about this then also add the link. Toolchain issues are incredibly time-consuming so the last thing they need is incomplete information.

I realize you're "only" moving an existing comment but as noted by @andyross this PR does far more than just re-organizing code.

Reply:
I think @dcpleung was the one who discovered that particular optimizer misfeature, likely on the MTL toolchain?

Reply:
All the xt-clang I used exhibited the same behavior of "ignoring" NOPs beyond 8.